src/client_side.cc

```diff
@@ -657,28 +656,26 @@ clientSetKeepaliveFlag(ClientHttpRequest * http)
 }
 
 /// checks body length of non-chunked requests
-static int
+static bool
 clientIsContentLengthValid(HttpRequest * r)
 {
     // No Content-Length means this request just has no body, but conflicting
     // Content-Lengths mean a message framing error (RFC 7230 Section 3.3.3 #4).
     if (r->header.conflictingContentLength())
-        return 0;
-
-    switch (r->method.id()) {
-
-    case Http::METHOD_GET:
+        return false;
 
-    case Http::METHOD_HEAD:
-        /* We do not want to see a request entity on GET/HEAD requests */
-        return (r->content_length <= 0 || Config.onoff.request_entities);
+    // RFC 7230 section 3.3 - Content-Length defines body exists regardless of method
+    if (r->content_length > 0)
+        return true;
 
-    default:
-        /* For other types of requests we don't care */
-        return 1;
-    }
+    // We do not want a request entity on HTTP/1.0 GET/HEAD requests
+    // without a Content-Length header.
+    const auto m = r->method.id();
+    if (m == Http::METHOD_GET || m == Http::METHOD_HEAD)
+        return !(r->http_ver <= Http::ProtocolVersion(1,0));
```

Comment:
Please reword the comment to explain "why" we added this special case instead of just retelling what the code does in English.

Reply:
The reason is vague reference in the direction of "cache corruption" in messages without Content-Length header in the RFC document. I've not found any specific CVE issue, though I presume the issue relates to when the broken Keep-Alive header was used (not an issue in HTTP/1.1). So the reason is as stated "we do not want" for those old versions - which are quite rare now.

Reply:
Your response does not address my concern. FWIW, I will document the same concern from another angle: Right now, this PR code rejects HTTP/1.0 GET requests without Content-Length/Transfer-Encoding headers. I am asking you to explain why they should be rejected. I do not know which "vague reference ... in the RFC document" you are talking about, but this drastic rejection deserves a much better explanation than vague references to vague references.

FTR, after covering cases with Content-Length and Transfer-Encoding headers (irrelevant to the specific code being discussed in this particular change request), RFC 7230 says: `If this is a request message and none of the above are true, then the message body length is zero (no message body is present).`

The RFC position is clear/natural/simple -- if there is no signal that request has a body, then the request does not have a body. This position is compatible with real traffic/implementations because "nobody" sends Content-Length:0 with requests that do not usually have a body (like GET or DELETE) and nobody wants to add a lot of custom/complex code to handle a rare (and unnecessary) case of a request that is successfully terminated by its transport connection closure.

We may want to treat some exceptional cases specially, of course, but we need to explain why. I speculate that something went wrong when this PR tried to detect and/or treat some exceptional cases, but the existing documentation/comments do not help me guess what went wrong.

Reply:
Oh. I think I see what you mean. I have a complete re-write of the sequence with better documentation coming shortly that should resolve this mistake and meet your other requests at the same time.

Reply:
Sounds good. Please request my review when you are done.

Reply:
done in fc31d38